src/components/structures/auth/CompleteSecurity.js

```diff
@@ -0,0 +1,173 @@
+            phase,
+        } = this.state;
+
+        let icon;
+        let title;
+        let body;
+        if (phase === PHASE_INTRO) {
+            icon = <span className="mx_CompleteSecurity_headerIcon mx_E2EIcon_warning"></span>;
+            title = _t("Complete security");
+            body = (
+                <div>
+                    <p>{_t(
+                        "Verify this session to grant it access to encrypted messages.",
+                    )}</p>
+                    <div className="mx_CompleteSecurity_actionRow">
+                        <AccessibleButton
+                            kind="danger"
+                            onClick={this.onSkipClick}
+                        >
+                            {_t("Skip")}
+                        </AccessibleButton>
+                        <AccessibleButton
+                            kind="primary"
+                            onClick={this.onStartClick}
+                        >
+                            {_t("Start")}
+                        </AccessibleButton>
+                    </div>
+                </div>
+            );
+        } else if (phase === PHASE_DONE) {
+            icon = <span className="mx_CompleteSecurity_headerIcon mx_E2EIcon_verified"></span>;
+            title = _t("Session verified");
+            body = (
+                <div>
+                    <div className="mx_CompleteSecurity_heroIcon mx_E2EIcon_verified"></div>
+                    <p>{_t(
+                        "Your new session is now verified. It has access to your " +
+                        "encrypted messages, and other users will see it as trusted.",
+                    )}</p>
+                    <div className="mx_CompleteSecurity_actionRow">
+                        <AccessibleButton
+                            kind="primary"
+                            onClick={this.onDoneClick}
+                        >
+                            {_t("Done")}
+                        </AccessibleButton>
+                    </div>
+                </div>
+            );
+        } else if (phase === PHASE_CONFIRM_SKIP) {
+            icon = <span className="mx_CompleteSecurity_headerIcon mx_E2EIcon_warning"></span>;
+            title = _t("Are you sure?");
+            body = (
+                <div>
+                    <p>{_t(
+                        "Without completing security on this device, it won’t have " +
+                        "access to encrypted messages.",
+                    )}</p>
+                    <div className="mx_CompleteSecurity_actionRow">
+                        <AccessibleButton
+                            className="warning"
+                            kind="secondary"
+                            onClick={this.onSkipConfirmClick}
+                        >
+                            {_t("Skip")}
+                        </AccessibleButton>
+                        <AccessibleButton
+                            kind="danger"
```

Comment:
it feels a bit odd to have the safe button be dangerous, but I presume this is from the design?

Reply:
Yeah, I was also a bit surprised, but seems intended from the design and replicated to other screens as well.

From Figma:

<img width="400" alt="2020-01-15 at 22 06" src="https://user-images.githubusercontent.com/279572/72475788-8a8fb880-37e3-11ea-92b3-66c48210c2e8.png">